tornjak-frontend/src/components/RenderOnAdminRole.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import TornjakHelper from './tornjak-helper';
+import AccessNotAllowed from './AccessNotAllowed'
+import { connect } from 'react-redux';
+import { RootState } from 'redux/reducers';
+
+type RenderOnAdminRoleProp = {
+    // updated user roles
+    globalUserRoles: string[],
+    children: React.ReactNode
+}
+
+type RenderOnAdminRoleState = {}
+
+class RenderOnAdminRole extends Component<RenderOnAdminRoleProp, RenderOnAdminRoleState> {
+    TornjakHelper: TornjakHelper;
+    constructor(props: RenderOnAdminRoleProp) {
+        super(props);
+        this.TornjakHelper = new TornjakHelper(props);
+        this.state = {};
+    }
+
+    checkPath() {
+        const pathsWithAdminRestr = [
+            "/entry/create",
+            "/agent/createjointoken",
+            "/cluster/clustermanagement"];
+        const isPath = pathsWithAdminRestr.find(element => {
+            if(element.includes(window.location.pathname)) {
+                return true;
+            }
+            return false;
+        });
+        return isPath;
+    }
```

Comment:
Cool function to put all the information in one place!
I wonder if this were to be used on other links/pages, would it apply this logic? It'd be great if it does, because then maybe App.tsx can be more uniform and wrap around all pages. That could potentially add flexibility if we want to make the auth logic more flexible

Reply:
unrelated, but associated with this few lines ... I'm not sure, but another approach to explore here is also having this be done on a div tag level (or jsx equivalent), something to consider. The nature of access control is endpoint based and so it may be difficult to have everything be bound to this location...

Reply:
@maia-iyer yes this logic applies to all routes. Same as backend logic these are the restricted routes for admin so checks these against all links. And App.tsx wraps around all routes.